packages/block-library/src/navigation-menu/edit.js

```diff
@@ -108,14 +113,70 @@ function NavigationMenu( {
 		}
 	};
 
-	useEffect( () => {
-		// Set/Unset colors CSS classes.
-		setAttributes( {
-			backgroundColorCSSClass: backgroundColor.class ? backgroundColor.class : null,
-			textColorCSSClass: textColor.class ? textColor.class : null,
-		} );
-	}, [ backgroundColor.class, textColor.class ] );
+	const handleCreateEmpty = () => {
+		setBlocksTemplate( null );
+		setInitialPlaceholder( false );
```

Comment:
Why do we need all this state handling for the placeholder? It should just be a reflection of the state of other attributes:
- If block is empty (has no inner-blocks) => show placeholder.
- If block has only one inner block and it's empty => corresponds to start from scratch state.
- If block has multiple inner block elements => normal state.

Reply:
The placeholder showing can be a function of other props yes.

However, the buttons need to change the state of `innerBlocks`'s `template` prop.

 I needed someway to change the state of the component to tell `innerBlocks` what template it was going to use depending on which button was clicked:

1. No template.
2. Pages (which I'm going to load on demand shortly).

We can ditch the `[initialPlaceholder,setInitialPlaceholder]` but I felt we'd need the `blocksTemplate` especially when we'll offer different types of template in future (eg: Pages, Menu items...etc).

Reply:
States:

1. Placeholder
2. Created from Top Level Items
3. Created Empty

The `template` prop passed to `<innerBlocks>` determines the state of the UI. We change the `template` prop based on which option the user clicks. If there are existing innerBlocks we simply pass `null` as the `template` prop which allows the existing Blocks to appear.

Reply:
> I felt we'd need the blocksTemplate especially when we'll offer different types of template in future (eg: Pages, Menu items...etc).

When the time comes to offer those, the model can be expanded. Until then, I expect this to hold:

>  It should just be a reflection of the state of other attributes

including in:

> I needed someway to change the state of the component to tell innerBlocks what template it was going to use depending on which button was clicked

Reply:
As I suspected (and failed to communicate effectively) this wasn't as simple as indicated above.

I discussed with @mtias and agreed that state was non-optimal in a Block context. I agree with this entirely and now better understand the reasons why.

However the issue of handling `InnerBlocks` was less simple. Our use case is unlike Media+Text or Columns because we do not utilise the `TemplateOptions` feature of `InnerBlocks` to render our placeholder. Whilst our version _looks_ visually similar, we have to render a `<Placeholder />` directly because `InnerBlocks` doesn't support a true placeholder attribute. As a result, we have a conditional render of either:

1. `<Placeholder />` - `InnerBlocks` is _not_ rendered here. We return early from render.
2. Block UI (ie: the Nav Block with all the controls...etc) - this is where `InnerBlocks` is rendered.

My point was that in the first case we do not render `InnerBlocks. As a result we need a way to change the "state" (not React `state`) of the component from `1` -> `2` when one of the `Button`s are clicked in the placeholder UI.

After discussing this at length, we decided that `dispatch`ing `replaceInnerBlocks` in the Button `onClick` handlers was the best way. This would trigger an update to `innerBlocks` causing `hasInnerBlocks` to return `true` therefore allowing the UI to transition from `1` -> `2`.

Apologies that I didn't communicate myself effectively. I've now updated the implementation and it's working without `useState` using the method described above.

Reply:
See updated PR desc screencapture.